sites/game-client/ui/overlay.mjs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2021 HITCON Online Contributors
+// SPDX-License-Identifier: BSD-2-Clause
+
+// enum
+const OverlayPosition = Object.freeze({
+    LEFT_TOP: Symbol(),
+    LEFT_CENTER: Symbol(),
+    LEFT_BOTTOM: Symbol(),
+    RIGHT: Symbol()
+    //CENTER_TOP: Symbol(), // overlapped with notification
+    /* 
+    CENTER_BOTTOM: Symbol(),
+    RIGHT_TOP: Symbol(),
+    RIGHT_CENTER: Symbol(),
+    RIGHT_BOTTOM: Symbol()
+    */
+});
+
+/** 
+ * Represents an overlay
+ * 
+ * Those who wants to use the Overlay element should extend this class and implement the methods.
+*/
+class Overlay {
+    /**
+     * 
+     * @param dom HTML DOM element.
+     */
+    constructor(dom) {}
+  
+    // Overlay hide/show related properties.
+    canHide() {}
```

Comment:
Maybe write it as a jsdoc? Also, it is supposed to be implemented by the derived class right?